test/test_common/test_time.h

```diff
@@ -21,28 +22,41 @@ class TestRealTimeSystem : public TestTimeSystem {
   }
 
   // TimeSource
-  SystemTime systemTime() override { return real_time_system_.systemTime(); }
-  MonotonicTime monotonicTime() override { return real_time_system_.monotonicTime(); }
+  SystemTime systemTime() override;
+  MonotonicTime monotonicTime() override;
 
 private:
   Event::RealTimeSystem real_time_system_;
 };
 
+class GlobalTimeSystem : public DelegatingTestTimeSystemBase<TestTimeSystem> {
+public:
+  TestTimeSystem& timeSystem() override {
+    if (singleton_->timeSystem() == nullptr) {
+      // TODO(jmarantz): Switch default to SimulatedTimeSystem.
```

Comment:
Minor preference for TODOs with issue numbers rather than usernames.

Reply:
That's true. But you could just tag "no stalebot" for anything that was a long term cleanup like this. Up to you (I'd insist on bug number internally, but I don't feel strongly for Envoy).